packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -1176,6 +1200,20 @@ export const formNestedStack = async (
         }
         if (resourceDetails.providerMetadata) {
           templateURL = resourceDetails.providerMetadata.s3TemplateURL;
+          const urlParameters = urlParametersMap[templateURL];
+
+          if (urlParameters.length) {
+            const nonCustomParameters = ['CloudWatchRule', 'deploymentBucketName', 'env', 's3Key'];
+            const misMatchedParams = Object.keys(parameters).filter(
+              (parameter) => !nonCustomParameters.includes(parameter) && !urlParameters.includes(parameter),
+            );
+            if (misMatchedParams.length) {
```

Comment:
Iff there's a mismatch here, it is almost certainly a Fault, not an Error. A custom resource is probably the only case that should be classified as an Error. Also, what about parameters in the child stack that have default values? These parameters don't need values to be specified by the parent stack